### PR TITLE
Optimize OTel Trace ID queries with skip index

### PR DESCRIPTION
**Pull request by SpencerTorres**
Adds the optimization requested in #724 

This optimization is only included in the SQL when OTel is enabled since the table is specific to OTel.

Here's the new SQL that is generated for Trace ID queries:

```diff
+ WITH
+ '207c6fb96eda8541559a283678587c86' as trace_id,
+ (SELECT min(Start) FROM "default"."otel_traces_trace_id_ts" WHERE TraceId = trace_id) as trace_start,
+ (SELECT max(End) + 1 FROM "default"."otel_traces_trace_id_ts" WHERE TraceId = trace_id) as trace_end
SELECT
"TraceId" as traceID, "SpanId" as spanID, "ParentSpanId" as parentSpanID, "ServiceName" as serviceName,
"SpanName" as operationName, "Timestamp" as startTime, multiply("Duration", 0.000001) as duration,
arrayMap(key -> map('key', key, 'value',"SpanAttributes"[key]), mapKeys("SpanAttributes")) as tags,
arrayMap(key -> map('key', key, 'value',"ResourceAttributes"[key]), mapKeys("ResourceAttributes")) as serviceTags
FROM "default"."otel_traces"
WHERE
- traceID = '207c6fb96eda8541559a283678587c86'
+ traceID = trace_id
+ AND startTime >= trace_start
+ AND startTime <= trace_end 
LIMIT 1000
```

Here's a rough graph of query duration before and after:
image

### CODE REVIEW

File: src/otel.ts [hunk 1]
```diff
@@ -56,3 +59,10 @@ export const versions: readonly OtelVersion[] = [
 ];
 
 export const getLatestVersion = (): OtelVersion => versions[0];
+export const getVersion = (version: string | undefined): OtelVersion | undefined => {
```
Comment (asimpson) [inline src/otel.ts:62]: This is a major nit but should we rename this function to `getOtelVersion`? We use `import { versions as otelVersions}` exclusively and I wonder if we should rename that to `otelVersions` as well?

File: src/data/sqlGenerator.ts [hunk 2]
```diff
@@ -145,24 +146,46 @@ const generateTraceIdQuery = (options: QueryBuilderOptions): string => {
   }
   const selectPartsSql = selectParts.join(', ');
 
+  // Optimize trace ID filtering for OTel enabled trace lookups
+  const hasTraceIdFilter = options.meta?.isTraceIdMode && options.meta?.traceId;
+  const otelVersion = getOtelVersion(options.meta?.otelVersion);
+  const applyTraceIdOptimization = hasTraceIdFilter && options.meta?.otelEnabled && otelVersion;
+  if (applyTraceIdOptimization) {
+    const traceId = options.meta!.traceId;
```
Comment (asimpson) [inline src/data/sqlGenerator.ts:154]: 🤘 😏

**Review by asimpson — APPROVED** [review]
LGTM 🎉 

Just two nits, up to you on how you want to handle them 😎

**Comment by SpencerTorres** [comment]
Good points on the otel functions, pushed a change to clean up the imports/references 👌

File: src/otel.ts [hunk 3]
```diff
@@ -56,3 +59,16 @@ export const versions: readonly OtelVersion[] = [
...
+  if (!version) {
+    return;
+  }
+
+  return versions.find(v => v.version === version);
+};
+
+export default {
```
Comment (asimpson) [inline src/otel.ts:70]: :heart_eyes:

**Review by asimpson — APPROVED** [review]
:trophy: nicely done

**Comment by tanner-bruce** [comment]
FYI this query scans the trace_id_ts table twice.

Compare with:
```sql
(SELECT (min(start), max(end) + 1) FROM tracing.otel_traces_trace_id_ts WHERE trace_id = ID) as bounds,
...
AND startTime >= bounds.1
AND startTime <= bounds.2 
```

**Comment by SpencerTorres** [comment]
> FYI this query scans the trace_id_ts table twice.

Very good point, this PR was simply using the linked query on the original issue, but this seems like another optimization we can add. Thanks!

**Pull request merged**